tests/qchem/test_observable_hf.py

```diff
@@ -236,10 +238,49 @@ def test_fermionic_observable(core_constant, integral_one, integral_two, f_ref):
 )
 def test_qubit_observable(f_observable, q_observable):
     r"""Test that qubit_observable returns the correct operator."""
-    h = qchem.qubit_observable(f_observable)
+    h_as_hamiltonian = qchem.qubit_observable(f_observable)
     h_ref = qml.Hamiltonian(q_observable[0], q_observable[1])
 
-    assert h.compare(h_ref)
+    enable_new_opmath()
+
+    h_as_op = qchem.qubit_observable(f_observable)
+    h_ref_as_op = pauli_sentence(h_ref).operation()  # easy conversion from ham to operation
+
+    disable_new_opmath()
+
+    assert h_as_hamiltonian.compare(h_ref)
+    assert np.allclose(
+        qml.matrix(h_as_op, wire_order=[0, 1, 2]), qml.matrix(h_ref_as_op, wire_order=[0, 1, 2])
+    )
```

Comment:
Why not just modify the tests to make them compatible with the new return type? Is this format needed to enable deprecating `Hamiltonian` all at once across the whole PL?

Reply:
Yes exactly, we still support using `Hamiltonian` until we properly deprecate it